### PR TITLE
Correctly Support Translating External Ids

### PR DESCRIPTION
**Pull request by mark-monteiro**
This builds upon #2643 adding a few additional changes. The changes made since that PR are summarized below:
- For the external id types, remove the `ExternalIdMediaType` `None` option and instead represent a general external id type with a value of 'null'
- Move `ExternalIdMediaType` to `MediaBrowser.Model` and use it directly in the model class instead of converting to a string
- Rename `Name` to `ProviderName` to more accurately reflect the new purpose of that property

The original description from #2643 is also copied below

---

**Changes**
When we have multiple implementations of `IExternalId` for the same provider, there is no way to distinguish the different IDs on the UI. Some types have handled this by adding the sub type to the `Name` property but this doesn't allow for localization. This change ads a `Type` Property which allows the specific type to be specified and displayed in the UI. This is a different approach than the previous PR  (#2236) which handled the localization on the server. If we decide to go with this PR, I will close the previous one.

There is a separate PR for the jellyfin-client changes which I will link here once it is created.

**Issues**
jellyfin/jellyfin-web#808
jellyfin/jellyfin-web#900

### CODE REVIEW

File: MediaBrowser.Model/Providers/ExternalIdMediaType.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
+namespace MediaBrowser.Model.Providers
+{
+    /// <summary>
+    /// The specific media type of an <see cref="ExternalIdInfo"/>.
+    /// </summary>
+    /// <remarks>
+    /// Client applications may use this as a translation key.
+    /// </remarks>
+    public enum ExternalIdMediaType
```
Comment (crobibero) [inline MediaBrowser.Model/Providers/ExternalIdMediaType.cs:9]: Should we be setting the Enum value here?

Comment (mark-monteiro) [reply MediaBrowser.Model/Providers/ExternalIdMediaType.cs:9]: I've added explicit enumeration values here

**Comment by mark-monteiro** [comment]
@crobibero FYI I made an additional change to this PR. I've removed the 'General' enum value and instead made the 'Type' property nullable so we can set the value to null for general external id types. This is preferable because it makes translation client-side easier since there is no need to check for the 'General' enum value (I've closed the associated web PR since it's no longer needed)

**Comment by lfoust** [comment]
Looks good. Thanks @mark-monteiro!

**Pull request merged**